### PR TITLE
Syft load based ONLY on rank

### LINKED ISSUE #3043: Create a syft.load(tensor, src=worker_id) functionality
This should be equivalent to worker.search, but is supposed to be retrieving information _locally_, when src matches the current worker. (behaviour is supposed to be close to crypten.load)
As is, this function is supposed not to fail and to always return something  (even random) when using in a plan at building.
Details on implementation are not all solved yet.

### PR DESCRIPTION
**Pull request by gmuraru**
Fixes #3043 
Send the party information in the init message such that we need to keep track only of the worker id.

### CODE REVIEW

File: syft/workers/base.py [hunk 1]
```diff
@@ -970,8 +1012,13 @@ def search(self, query: Union[List[Union[str, int]], str, int]) -> List[PointerT
                 # set garbage_collect_data to False because if we're searching
                 # for a tensor we don't own, then it's probably someone else's
                 # decision to decide when to delete the tensor.
-                ptr = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker).wrap()
-                results.append(ptr)
+                if obj.owner is not self:
```
Comment (gmuraru) [inline syft/workers/base.py]: If the worker is the owner of the object - then simply return the object

Comment (karlhigley) [reply syft/workers/base.py]: This plays a bit fast and loose with types (tensors vs pointers), which works fine in Python but becomes kind of a pain to encode in the Syft protocol and deal with in other languages. If this is a convenience change that isn't required for the main purpose of the PR, I'd probably rather leave part alone.

**Comment by gmuraru** [comment]
This is a PR with the changes + what was added on the Master Branch (to keep it up to date).
The only effective changes were made in:
```
syft/frameworks/crypten/__init__.py
syft/frameworks/crypten/context.py
syft/workers/base.py
```

**Comment by karlhigley** [comment]
@gmuraru This is now triggering a bunch of translation tests, since it (technically) makes changes to a bunch of the translation paths. You might want to locally merge `master` into `pytorch/crypten` and push that directly to `pytorch/crypten` first, then rebase this PR on top.

File: syft/workers/base.py (hunk 1 above)
Comment (gmuraru) [reply syft/workers/base.py]: This PR does not depend necessary on this (I can call ```get``` in the ```load function```).
I think the problem was more "semantics" wise - like if you have the tensor  - why you should get a pointer to that tensor and not the effective tensor?

Comment (karlhigley) [reply syft/workers/base.py]: To make the return type consistent. If sometimes I get one thing and sometimes I get another thing, then I have to check what was returned in order to know how to process it, which proliferates `if` statements through the code base.

Comment (gmuraru) [reply syft/workers/base.py]: Yep, will update this.

File: syft/frameworks/crypten/__init__.py [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
-def load(tag: str, src: int, id_worker: int):
-    if comm.get().get_rank() == src:
-        worker = syft.local_worker.get_worker(id_worker)
-        result = worker.search(tag)[0].get()
+def load(tag: str, src: int):
+    worker_id = syft.local_worker.get_worker_id_from_rank(src)
+    assert worker_id != -1
+
+    try:
+        worker = syft.local_worker.get_worker(worker_id, fail_hard=True)
```
Comment (youben11) [inline syft/frameworks/crypten/__init__.py]: we should keep crypten's paradigm of checking `src` value for differentiating the operations to be done, some issues may if the `get_worker` return a worker for a non-expected `worker_id`

File: syft/frameworks/crypten/context.py [hunk 3]
```diff
@@ -133,8 +139,8 @@ def wrapper(*args, **kwargs):
             # Send messages to other workers so they start their parties
             threads = []
             for i in range(len(workers)):
-                rank = i + 1
-                msg = CryptenInit((rank, world_size, master_addr, master_port))
+                rank = i+1
```
Comment (youben11) [inline syft/frameworks/crypten/context.py]: spaces

Comment (gmuraru) [reply syft/frameworks/crypten/context.py]: Done

File: syft/frameworks/crypten/__init__.py (hunk 2 above)
Comment (gmuraru) [reply syft/frameworks/crypten/__init__.py]: I modified to use the rank :)

File: syft/frameworks/crypten/__init__.py [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
 
 
-def load(tag: str, src: int, id_worker: int):
-    if comm.get().get_rank() == src:
-        worker = syft.local_worker.get_worker(id_worker)
-        result = worker.search(tag)[0].get()
+def load(tag: str, src: int):
+    worker = syft.local_worker.get_worker_from_rank(src)
+
+    if worker and src == worker.rank:
```
Comment (youben11) [inline syft/frameworks/crypten/__init__.py]: Can't we simply check either `src == comm.get().get_rank()` ?

Comment (gmuraru) [reply syft/frameworks/crypten/__init__.py]: Yep...you are right. I overcomplicated this :(

**Comment by gmuraru** [comment]
will close this one since we will use tags for retrieving data.

**Pull request closed without merging**